### PR TITLE
Feat/damage health system

### PR DESCRIPTION
**Pull request by NathanGr33n**
Damage and Health working. Display's weapon image in corner of screen. Will add some more comments tomorrow.

### CODE REVIEW

File: TKGame/BackEnd/EntityManager.cs [hunk 1]
```diff
@@ -89,7 +90,33 @@ public static void Update(GameTime gameTime)
...
+                entity.healthTexture.SetData(new Color[] { Color.Red });
+                spriteBatch.Draw(entity.healthTexture, entity.healthBar, Color.White);
+            }
+        }
+
+        public static void DamageEnemy()
+        {
+            foreach (Entity entity in entities)
+            {
+                if (entities[0].hitBox.Intersects(entity.hitBox) && entity != entities[0])
```
Comment (NateCastleOIT) [inline TKGame/BackEnd/EntityManager.cs:114]: How come entities[0] is specified here? Are you trying to access the player? Player.Instance for that or query the list for the player instead so this doesn't potentially try to call on an entity other than what is intended

File: TKGame/Entity.cs [hunk 2]
```diff
@@ -22,6 +22,12 @@ public abstract class Entity : CollideComponent
         public string entityName; // to identify each entity by name
 
         public Weapon weapon;
+        public int health;
+        public Rectangle healthBar;
+        public Texture2D healthTexture;
+        public bool needsHealth = false;
```
Comment (aaronw84) [inline TKGame/Entity.cs:28]: What is needsHealth?

**Review by NateCastleOIT — APPROVED** [review]
LG, will just need to merge with power up system (Primary powerup basically) which can be next sprint

**Review by aaronw84 — APPROVED** [review]
Could use more comments, but if it runs it's good for demo.

**Pull request merged**